File: spring-batch-java/src/main/java/spring/batch/java/validator/TargetDateValidator.java
package spring.batch.java.validator;

import java.time.LocalDate;
import java.time.format.DateTimeParseException;

import org.springframework.batch.core.JobParameters;
import org.springframework.batch.core.JobParametersInvalidException;
import org.springframework.batch.core.JobParametersValidator;
import org.springframework.util.StringUtils;

public class TargetDateValidator implements JobParametersValidator {

	private static final String PARAMETER_NAME = "targetDate";

	@Override
	public void validate(JobParameters parameters) throws JobParametersInvalidException {
		String targetDate = parameters.getString(PARAMETER_NAME);

		if (!StringUtils.hasText(targetDate)) {
			throw new JobParametersInvalidException("%s 가 빈 문자열이거나 존재하지 않습니다.".formatted(PARAMETER_NAME));
		}

		try {
			LocalDate.parse(targetDate);
		} catch (DateTimeParseException e) {
			throw new JobParametersInvalidException("%s 가 날짜 형식의 문자열이 아닙니다.".formatted(PARAMETER_NAME));
		}
	}
}

File: spring-batch-kotlin/src/main/kotlin/spring/batch/kotlin/validator/TargetDateValidator.kt
package spring.batch.kotlin.validator

import org.springframework.batch.core.JobParameters
import org.springframework.batch.core.JobParametersInvalidException
import org.springframework.batch.core.JobParametersValidator
import java.time.LocalDate

class TargetDateValidator : JobParametersValidator {

    companion object {
        private const val PARAMETER_NAME = "targetDate"
    }

    override fun validate(parameters: JobParameters?) {
        val targetDate = parameters?.getString(PARAMETER_NAME).orEmpty()

        if (targetDate.isBlank()) {
            throw JobParametersInvalidException("$PARAMETER_NAME 가 빈 문자열이거나 존재하지 않습니다.")
        }

        runCatching { LocalDate.parse(targetDate) }
            .onFailure { throw JobParametersInvalidException("$PARAMETER_NAME 가 날짜 형식의 문자열이 아닙니다.") }
    }
}